test-network-function/networking/suite.go

```diff
@@ -275,3 +308,110 @@ func testNodePort(env *config.TestEnvironment) {
 		}
 	})
 }
+
+func getDeclaredPortList(command string, containerNum int, podName, podNamespace string) []PortDeclared {
+	var result []PortDeclared
+	ocCommandToExecute := fmt.Sprintf(command, podName, podNamespace, containerNum)
+	res := utils.ExecuteCommand(ocCommandToExecute, ocCommandTimeOut, interactive.GetContext(expectersVerboseModeEnabled), func() {
+		log.Error("can't run command: ", command)
+	})
+	x := strings.Split(res, "\n")
+	for _, i := range x {
+		fmt.Println(i)
+		if strings.Contains(i, "containerPort") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			port = s[0]
+		}
+		if strings.Contains(i, "name") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			name = s[0]
+		}
+		if strings.Contains(i, "protocol") {
+			s := strings.Split(i, ":")
+			protocolName = s[1]
+			flag = 1
+		}
+		if flag < 1 {
+			continue
+		}
+		p, _ := strconv.Atoi(strings.TrimSpace(port))
+		noderes := PortDeclared{containerPort: p, name: name, protocol: protocolName}
+		result = append(result, noderes)
+		flag = 0
+	}
+	return result
+}
+
+func getListeningPortList(ocCommand string) []PortListen {
```

Comment:
Might I suggest returning `([]PortListen, protocolName)`?  So we don't have have to use global variables for this.

Reply:
I didn't need to return here the variable protocol Name because this variable included in struct []PortListen.